addons/account/models/account_move.py

```diff
@@ -380,16 +380,23 @@ def _check_lock_date(self):
     def assert_balanced(self):
         if not self.ids:
             return True
-        prec = self.env.user.company_id.currency_id.decimal_places
-
-        self._cr.execute("""\
-            SELECT      move_id
-            FROM        account_move_line
-            WHERE       move_id in %s
-            GROUP BY    move_id
-            HAVING      abs(sum(debit) - sum(credit)) > %s
-            """, (tuple(self.ids), 10 ** (-max(5, prec))))
-        if len(self._cr.fetchall()) != 0:
+
+        # /!\ As this method is called in create / write, we can't make the assumption the computed stored fields
+        # are already done. Then, this query MUST NOT depend of computed stored fields (e.g. balance).
+        # It happens as the ORM makes the create with the 'no_recompute' statement.
+        self._cr.execute('''
+            SELECT line.move_id
+            FROM account_move_line line
+            JOIN account_move move ON move.id = line.move_id
+            JOIN account_journal journal ON journal.id = move.journal_id
+            JOIN res_company company ON company.id = journal.company_id
+            JOIN res_currency currency ON currency.id = company.currency_id
```

Comment:
the company currency have a direct access by the field `company_currency_id` in `account_move_line`, and you can reduce the number of JOIN, no ?

`LEFT JOIN res_currency rc ON (rc.id = aml.company_currency_id)`

Reply:
You can't use a stored computed field there as this query is used inside the `create`

Reply:
It's also the reason why I made `debit - credit` instead of `balance`.

Reply:
Ok, I understand